core/app/helpers/spree/products_helper.rb

```diff
@@ -63,7 +63,8 @@ def line_item_description_text(description_text)
     end
 
     def cache_key_for_products(products = @products, additional_cache_key = nil)
-      max_updated_at = (products.maximum(:updated_at) || Date.today).to_s(:number)
+      products_max_updated_at = products.max_by(&:updated_at)&.updated_at
```

Comment:
@mad-eel this will not work, this effectively fetches all the product records, creates AR objects, and iterates through them - this will decrease the performance by a lot - `.maximum` on the other hand executes only one quick SQL query